src/replica/prepare_list.cpp

```diff
@@ -74,6 +74,7 @@ error_code prepare_list::prepare(mutation_ptr &mu,
     decree d = mu->data.header.decree;
     dcheck_gt_replica(d, last_committed_decree());
 
+    mu->tracer->add_point("start prepare");
```

Comment:
```suggestion
    ADD_POINT(mu->tracer);
```

I have a suggestion. We can define a macro to simplify the naming of a tracepoint.

```
#define ADD_POINT(tracer) (tracer).add_point(__PRETTY_FUNCTION__)
```

`__PRETTY_FUNCTION__` is a non-standard extension provided by both gcc and clang, not supported by vsc++. I tested that works on gcc 4.8.2. https://gcc.gnu.org/onlinedocs/gcc/Function-Names.html 

This macro will add a point: `"error_code prepare_list::prepare(mutation_ptr, partition_status::type, bool)"`. Much longer than `"start prepare"` though, but much clearer too.　

Another alternative is to combine `__FILE__`, `__LINE__` and `__FUNC__`. The formatted result could be　`prepare_list.cpp:77::prepare`. This is shorter, while clear enough.

Reply:
Firstly, the point usually contain more message, such as add [ip:port](https://github.com/XiaoMi/rdsn/pull/511/files#diff-f958f9a8962c3eab0f346345d574dfa1R287)

Secondly, the points actually do not all mean the `method` name, it may `add_point` at multi line which present different stage in one method, such as `replica::execute_mutation`: [point1](https://github.com/Shuo-Jia/rdsn/blob/3273929c93e9fdf1d6ee4c706447313fadc73e95/src/replica/replica.cpp#L226), [point2](https://github.com/Shuo-Jia/rdsn/blob/3273929c93e9fdf1d6ee4c706447313fadc73e95/src/replica/replica.cpp#L317)

In a word, `stage name` isn't equal with `method name`, we usually need describe the stage but not use `method` name

Reply:
Oh, but the suggestion is perfect, maybe I can let the `macro` support the `custom` message to add such `ip:port` message?

Reply:
I think `ADD_POINT` is a helper of `add_point()`. If you want a customized name of the point, use `add_point`. Otherwise, and normally use `ADD_POINT` instead.

Reply:
I just think formatted result `prepare_list.cpp:77::prepare` is very good, so I want every point has the format and can be added custom information sometime

Reply:
I have add macro:
```c++
#define ADD_POINT(tracer)                                                                          \
    (tracer)->add_point(fmt::format("{}:{}:{}", __FILENAME__, __LINE__, __FUNCTION__))
#define ADD_CUSTOM_POINT(tracer, message)                                                          \
    (tracer)->add_point(                                                                           \
        fmt::format("{}:{}:{}[{}]", __FILENAME__, __LINE__, __FUNCTION__, (message)))
```
and the result format as:
```
     ***************[TRACE:mutation[33]]***************
        TRACE:name=mutation.cpp:148:add_client_request                                   , span=                 708, total=                 708, ts=1598325215128566590
        TRACE:name=replica_2pc.cpp:139:init_prepare                                      , span=                1828, total=                2536, ts=1598325215128568418
        TRACE:name=prepare_list.cpp:78:prepare                                           , span=                2684, total=                5220, ts=1598325215128571102
        TRACE:name=replica_2pc.cpp:288:send_prepare_message[127.11.0.100:34803]          , span=                2590, total=                7810, ts=1598325215128573692
        TRACE:name=replica_2pc.cpp:288:send_prepare_message[127.11.0.100:34802]          , span=               36492, total=               44302, ts=1598325215128610184
        TRACE:name=mutation_log.cpp:56:append                                            , span=               21627, total=               65929, ts=1598325215128631811
        TRACE:name=mutation_log.cpp:132:commit_pending_mutations                         , span=                5573, total=               71502, ts=1598325215128637384
        TRACE:name=mutation_log.cpp:142:operator()[commit_pending_completed]             , span=               32096, total=              103598, ts=1598325215128669480
        TRACE:name=replica_2pc.cpp:501:on_append_log_completed                           , span=               11143, total=              114741, ts=1598325215128680623
        TRACE:name=replica_2pc.cpp:328:do_possible_commit_on_primary                     , span=                1075, total=              115816, ts=1598325215128681698
        TRACE:name=replica_2pc.cpp:559:on_prepare_reply[127.11.0.100:34803]              , span=              141244, total=              257060, ts=1598325215128822942
        TRACE:name=replica_2pc.cpp:559:on_prepare_reply[127.11.0.100:34802]              , span=               10748, total=              267808, ts=1598325215128833690
        TRACE:name=replica_2pc.cpp:328:do_possible_commit_on_primary                     , span=                2397, total=              270205, ts=1598325215128836087
        TRACE:name=replica.cpp:227:execute_mutation                                      , span=                 924, total=              271129, ts=1598325215128837011
        TRACE:name=replication_app_base.cpp:476:apply_mutation                           , span=                 921, total=              272050, ts=1598325215128837932
        TRACE:name=replica.cpp:318:execute_mutation[completed]                           , span=               63593, total=              335643, ts=1598325215128901525
```